File: Count Subarrays With Score Less Than K/Count Subarrays With Score Less Than K.py
# Solution for Count Subarrays With Score Less Than K in PY

class Solution:
    def countSubarrays(self, nums: List[int], k: int) -> int:
        n, i, j, res, sum = len(nums), 0, 0, 0, 0
        while j < n:
            sum += nums[j]
            while i <= j and sum * (j - i + 1) >= k:
                sum -= nums[i]
                i += 1
            res += (j - i + 1)
            j += 1
        return res

File: Count Subarrays With Score Less Than K/Count Subarrays With Score Less Than K.cpp
// Solution for Count Subarrays With Score Less Than K in CPP

class Solution {
public:
    long long countSubarrays(vector<int>& nums, long long k) {
        int n = nums.size();
        int i = 0, j = 0;
        long long res = 0, sum = 0;
        while (j < n) {
            sum += nums[j];
            while (i <= j && sum * (j - i + 1) >= k) {
                sum -= nums[i];
                i++;
            }
            res += (j - i + 1);
            j++;
        }
        return res;
    }
};
